ingestion/src/metadata/orm_profiler/profiler/sampler.py

```diff
@@ -51,14 +58,20 @@ def __init__(
 
     @partition_filter_handler(build_sample=True)
     def get_sample_query(self) -> Query:
-        return (
-            self.session.query(
-                self.table, (ModuloFn(RandomNumFn(), 100)).label(RANDOM_LABEL)
-            )
-            .suffix_with(
-                f"SAMPLE BERNOULLI ({self.profile_sample or 100})",
-                dialect=Dialects.Snowflake,
+        if self.profile_sample_type == ProfileSampleType.PERCENTAGE:
+            return (
+                self.session.query(
+                    self.table, (ModuloFn(RandomNumFn(), 100)).label(RANDOM_LABEL)
+                )
+                .suffix_with(
+                    f"SAMPLE BERNOULLI ({self.profile_sample or 100})",
+                    dialect=Dialects.Snowflake,
+                )
+                .cte(f"{self.table.__tablename__}_rnd")
             )
+        return (
+            self.session.query(self.table)
+            .limit(self.profile_sample)
             .cte(f"{self.table.__tablename__}_rnd")
```

Comment:
I believe you should be able to remove the `.cte` here since we won't do any further operation on this table and you are creating the cte line 102 for this sampling type.

Reply:
@ayush-shah this was not fixed

Reply:
without the cte, i am not able to get `inspect(rnd).c`
it gives me an error as it's not able to find `c`